list_symbols.vala

```diff
@@ -24,14 +26,24 @@ class Vls.ListSymbols : Vala.CodeVisitor {
         return all_syms;
     }
 
-    public DocumentSymbol? add_symbol (Vala.Symbol sym, SymbolKind kind) {
-        var current_sym = containers.is_empty ? null : containers.peek_head ();
-        var dsym = new DocumentSymbol.from_vala_symbol (sym, kind);
+    public DocumentSymbol? add_symbol (Vala.Symbol sym, SymbolKind kind, bool adding_parent = false) {
+        var current_sym = (containers.is_empty || adding_parent) ? null : containers.peek_head ();
+        DocumentSymbol? dsym;
+        string sym_full_name = sym.get_full_name ();
+        bool unique = true;
+
+        if (sym is Vala.Namespace && ns_name_to_dsym.has_key (sym_full_name)) {
+            dsym = ns_name_to_dsym [sym_full_name];
+            unique = false;
```

Comment:
This is because `Vala.Namespace` symbols with the same text can appear multiple times in a file. There are cases like:

```
class Ns.Thing1 {}
class Ns.Thing2 {}
```

That lead to this.